File: Symfony/src/BGR/Serrano/ProductoBundle/Entity/UnidadDeMedida.php
<?php

namespace BGR\Serrano\ProductoBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation\Type;
use Doctrine\Common\Collections\ArrayCollection;
use JMS\Serializer\Annotation\Exclude;

/**
 * UnidadDeMedida
 *
 * @ORM\Table(uniqueConstraints={@ORM\UniqueConstraint(name="search_idx", columns={"name"})})
 * @ORM\Entity(repositoryClass="BGR\Serrano\ProductoBundle\Entity\UnidadDeMedidaRepository")
 * @ORM\HasLifecycleCallbacks()
 */
class UnidadDeMedida
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     *
     * @Type("integer")
     *
     */
    protected $id;

    /**
     * @var string
     *
     * @ORM\Column(name="name", type="string", length=50)
     *
     * @Type("string")
     *
     */
    protected $name;

    /**
     * @var string
     *
     * @ORM\Column(name="descripcion", type="string", length=50)
     *
     * @Type("string")
     *
     */
    protected $descripcion;

    /**
     * @var float
     *
     * @ORM\Column(name="equivalencia", type="float")
     *
     * @Type("float")
     *
     */
    protected $equivalencia;
  
    /**
     * @ORM\ManyToMany(targetEntity="Producto", mappedBy="unidad_de_medidas")
     * @Type("ArrayCollection<BGR\Serrano\ProductoBundle\Entity\Producto>")
     * @Exclude
     */
    private $productos;

    public function setProductos(\BGR\Serrano\ProductoBundle\Entity\Producto $productos)
    {
        $this->productos = $productos;
    }

    public function getProductos()
    {
        return $this->productos;
    }

    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
    }
    /**
     * Set name
     *
     * @param string $name
     * @return UnidadDeMedida
     */
    public function setName($name)
    {
        $this->name = $name;
    
        return $this;
    }

    /**
     * Get name
     *
     * @return string 
     */
    public function getName()
    {
        return $this->name;
    }
    /**
     * Set descripcion
     *
     * @param string $descripcion
     * @return UnidadDeMedida
     */
    public function setDescripcion($descripcion)
    {
        $this->descripcion = $descripcion;
    
        return $this;
    }

    /**
     * Get descripcion
     *
     * @return string 
     */
    public function getDescripcion()
    {
        return $this->descripcion;
    }

    /**
     * Set equivalencia
     *
     * @param float $equivalencia
     * @return UnidadDeMedida
     */
    public function setEquivalencia($equivalencia)
    {
        $this->equivalencia = $equivalencia;
    
        return $this;
    }

    /**
     * Get equivalencia
     *
     * @return float 
     */
    public function getEquivalencia()
    {
        return $this->equivalencia;
    }
    
    public function __construct()
    {
        $this->productos = new ArrayCollection();

    }
    	

    /**
     * Add productos
     *
     * @param \BGR\Serrano\ProductoBundle\Entity\Producto $productos
     * @return UnidadDeMedida
     */
    public function addProducto(\BGR\Serrano\ProductoBundle\Entity\Producto $producto)
    {
        $this->productos->add($producto);
        return $this;
    }

    /**
     * Remove productos
     *
     * @param \BGR\Serrano\ProductoBundle\Entity\Producto $productos
     */
    public function removeProducto(\BGR\Serrano\ProductoBundle\Entity\Producto $productos)
    {
        $this->productos->removeElement($productos);
    }
}

File: BGR/BGR-Data/src/main/java/ar/com/bgr/serrano/model/UnidadDeMedida.java
/**
 * 
 */
package ar.com.bgr.serrano.model;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.UniqueConstraint;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonDeserialize;
import org.codehaus.jackson.map.annotate.JsonSerialize;

/**
 * 
 * Descripcion:
 * 
 * {}
 *
 * @author matias
 * 
 * @since 01/06/2014
 */
@Entity(name="UnidadDeMedida")
@Table(name="UnidadDeMedida", uniqueConstraints = @UniqueConstraint(columnNames={"name"}))
@JsonIgnoreProperties(ignoreUnknown = true)
public class UnidadDeMedida {
	
	@Id
	@GeneratedValue
	private Integer id;
	
	@Column(name = "name")
	private String name;

	@Column(name = "descripcion")
	private String descripcion;	
	
	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "deriva_de_id")
	private UnidadDeMedida deriva_de;

	@Column(name = "divisible")
	private Boolean divisible;
	
	@Column(name = "equivalencia")
	private Double equivalencia;

	/**
	 * @return the id
	 */
	public Integer getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Integer id) {
		this.id = id;
	}

	/**
	 * @return the name
	 */
	public String getName() {
		return name;
	}

	/**
	 * @param name the name to set
	 */
	public void setName(String name) {
		this.name = name;
	}

	/**
	 * @return the descripcion
	 */
	public String getDescripcion() {
		return descripcion;
	}

	/**
	 * @param descripcion the descripcion to set
	 */
	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	/**
	 * @return the derivaDe
	 */
	public UnidadDeMedida getDerivaDe() {
		return deriva_de;
	}

	/**
	 * @param derivaDe the derivaDe to set
	 */
	public void setDerivaDe(UnidadDeMedida derivaDe) {
		this.deriva_de = derivaDe;
	}

	/**
	 * @return the divisible
	 */
	public Boolean getDivisible() {
		return divisible;
	}

	/**
	 * @param divisible the divisible to set
	 */
	public void setDivisible(Boolean divisible) {
		this.divisible = divisible;
	}

	/**
	 * @return the equivalencia
	 */
	public Double getEquivalencia() {
		return equivalencia;
	}

	/**
	 * @param equivalencia the equivalencia to set
	 */
	public void setEquivalencia(Double equivalencia) {
		this.equivalencia = equivalencia;
	}	
	
	

}
